spikeinterface/extractors/stream_ibl.py

```diff
@@ -0,0 +1,204 @@
+        ----------
+        session : str
+          DESCRIPTION.
+
+        Returns
+        -------
+        stream_names : list of str
+            List of stream names as expected by the `stream_name` argument for the class initialization.
+        """
+        from one.api import ONE
+
+        one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+
+        dataset_contents = one.list_datasets(eid=session, collection="raw_ephys_data/*")
+        raw_contents = [dataset_content for dataset_content in dataset_contents if not dataset_content.endswith(".npy")]
+        probe_labels = set([raw_content.split("/")[1] for raw_content in raw_contents])
+
+        stream_names = list()
+        for probe_label in probe_labels:
+            raw_suffixes_by_probe = set(
+                [Path(raw_content).suffixes[-2] for raw_content in raw_contents if "probe00" in raw_content]
+            )
+            if ".ap" in raw_suffixes_by_probe:
+                stream_names.append(probe_label + ".ap")
+            if ".lf" in raw_suffixes_by_probe:
+                stream_names.append(probe_label + ".lf")
+
+        return stream_names
+
+    def __init__(
+        self,
+        session: str,
+        stream_name: str,
+        load_sync_channel: bool = False,
+        cache_folder: Optional[Union[Path, str]] = None,
+        remove_cached: bool = True,
+    ):
+        from brainbox.io.spikeglx import Streamer
+        from one.api import ONE
+        from neo.rawio.spikeglxrawio import read_meta_file, extract_stream_info
+
+        one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+
+        session_names = self.get_stream_names(session=session)
+        assert stream_name in session_names, (
+            f"The `stream_name` '{stream_name}' was not found in the available listing for session '{session}'! "
+            f"Please choose one of {session_names}."
+        )
+        probe_label, stream_type = stream_name.split(".")
+
+        insertions = one.alyx.rest("insertions", "list", session=session)
+        pid = next(insertion["id"] for insertion in insertions if insertion["name"] == probe_label)
+
+        cache_folder = Path(cache_folder) if cache_folder is not None else cache_folder
+        file_streamer = Streamer(
+            pid=pid, one=one, typ=stream_type, cache_folder=cache_folder, remove_cached=remove_cached
+        )
+
+        # get basic metadata
+        meta_file = file_streamer.file_meta_data  # streamer downloads uncompressed metadata files on init
+        meta = read_meta_file(meta_file)
+        info = extract_stream_info(meta_file, meta)
+        channel_ids = info["channel_names"]
+        gains = info["channel_gains"]
+        offsets = info["channel_offsets"]
+        if not load_sync_channel:
+            channel_ids = channel_ids[:-1]
+            gains = gains[:-1]
+            offsets = offsets[:-1]
```

Comment:
might not be necessary, not sure more than one sync channels can be expected, but `file_streamer.nsync` would return the number of sync channels so that could be used here. 
```suggestion
            num_sync_channels = file_streamer.nsync
            channel_ids = channel_ids[:-num_sync_channels]
            gains = gains[:-num_sync_channels]
            offsets = offsets[:-num_sync_channels]
```

Reply:
I examined 60+ sessions and the value was never greater than 1

Reply:
I suspect this streamer does not load data from the NIDQ and we're instead supposed to use the other loaders for that